check_diff/src/main.rs

```diff
@@ -15,11 +20,36 @@ struct CliInputs {
     rustfmt_config: Option<Vec<String>>,
 }
 
-fn main() {
-    let args = CliInputs::parse();
+/// Clone a git repository and cd into it.
+///
+/// Parameters:
+/// url: git clone url
+/// dest: directory where the repo should be cloned
+fn clone_git_repo(url: &str, dest: &str) {
+    env::set_var("GIT_TERMINAL_PROMPT", "0");
+    let git_cmd = Command::new("git")
+        .args(["clone", "--quiet", url, "--depth", "1", dest])
+        .output()
+        .expect("failed to clone repository");
+    // if the git command does not return successfully,
+    // cd command will not work as well. So fail fast.
+    if !git_cmd.status.success() {
+        io::stdout().write_all(&git_cmd.stdout).unwrap();
+        io::stderr().write_all(&git_cmd.stderr).unwrap();
+        return;
+    }
+    println!("Successfully clone repository. Entering repository");
```

Comment:
Not something we have to change right now, but I'd love to replace these `println!` logs with [`tracing`](https://docs.rs/tracing/latest/tracing/) or the [`log`](https://docs.rs/log/latest/log/) crate. It probably doesn't make much of a difference, though we already use `tracing` within rustfmt.